CONTRIBUTING.md

```diff
@@ -0,0 +1,72 @@
+# Contributing
+
+Authorization Proxy uses GitHub to manage reviews of pull requests.
+
+- The versioning scheme we use is [SemVer](http://semver.org/).
+- Relevant coding style guidelines are the [Go Code Review Comments](https://code.google.com/p/go-wiki/wiki/CodeReviewComments) and the _Formatting and style_ section of Peter Bourgon's [Go: Best Practices for Production Environments](https://peter.bourgon.org/go-in-production/#formatting-and-style).
+
+## Steps to Contribute
+
+Should you wish to work on an issue, please claim it first by commenting on the GitHub issue that you want to work on it. This is to prevent duplicated efforts from contributors on the same issue.
+
+For quickly compiling and testing your changes, do:
+```bash
+# for building
+go build
+./authorization-proxy
+
+# for testing (Make sure all the tests pass before you commit and push :))
+make test
+```
+
+## Pull Request Process
+
+- Branch from the master branch and, if needed, rebase to the current master branch before submitting your pull request. If it doesn't merge cleanly with master you may be asked to rebase your changes.
+    - Branches should have descriptive names and start with prefixes like `patch/`, `fix/`, `feature/`. Good examples are: `fix/vulnerability-issue` or `feature/issue-templates`.
+- Commits should be as small as possible, while ensuring that each commit is correct independently (i.e., each commit should compile and pass tests).
```

Comment:
I think this is too harsh.

I think it should be met at stage of merging.

Reply:
I think it is better to keep in mind, just as a best practice.
We can always squash the commits into one before asking for review.🤪

Reply:
ok

Reply:
remove in 3502b8d